src/_pytest/compat.py

```diff
@@ -314,15 +314,24 @@ def safe_isclass(obj: object) -> bool:
 
 
 def get_user_id() -> int | None:
-    """Return the current user id, or None if we cannot get it reliably on the current platform."""
-    # win32 does not have a getuid() function.
-    # On Emscripten, getuid() is a stub that always returns 0.
-    if sys.platform in ("win32", "emscripten"):
+    """Return the current process's real user id or None if it cannot be
+    retrieved reliably.
+
+    :return: The user id or None
```

Comment:
```suggestion
    :return: The user id or None if it could not be determined.
```

Reply:
The rationale here is explaining why None could be returned, otherwise the description does not add much from the return-type we already have.